### PR TITLE
Modernizer plugin, Checkstyle warning fixes

### PR DESCRIPTION
**Pull request by stevenschlansker**
requires basepom/basepom#20

### CODE REVIEW

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/customizers/FetchDirection.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public SqlStatementCustomizer createForParameter(Annotation annotation,
     {
         private final Integer direction;
 
-        public FetchDirectionSqlStatementCustomizer(final Integer direction)
+        FetchDirectionSqlStatementCustomizer(final Integer direction)
```
Comment (qualidafial) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/customizers/FetchDirection.java]: This inner class could be a lambda in the caller

File: pom.xml [hunk 2]
```diff
@@ -18,7 +18,7 @@
     <parent>
         <groupId>org.basepom</groupId>
         <artifactId>basepom-standard-oss</artifactId>
-        <version>13</version>
+        <version>14-SNAPSHOT</version>
```
Comment (qualidafial) [inline pom.xml]: This is breaking the build. Travis can't find `14-SNAPSHOT`

**Comment by qualidafial** [comment]
One comment, not sure if related to the linked basepom PR.

This is good to merge once the TravisCI build is fixed.

**Comment by stevenschlansker** [comment]
Travis doesn't pick up snapshots I guess.  We can hold off on merging until basepom 14 is released probably, it's not exactly urgent.

**Comment by stevenschlansker** [comment]
I'm giving up on this.  It trips all over the guava and jodatime artifact, and it's a pain to configure it right.
Cherry-picked the fixups and dropped the plugin.

**Pull request closed without merging**